src/utils/FileBuffer.cpp

```
@@ -0,0 +1,285 @@
+    std::srand(std::time(0));
+  }
+
+  std::string filename;
+  filename.append("tmpfile_");
+  filename.append(ft::to_string(std::rand()));
+  filename.append(ft::to_string(std::rand()));
+  return filename;
+}
+// NOLINTEND(cert-msc30-c, cert-msc30-cpp, cert-msc32-c, cert-msc50-cpp,
+// cert-msc51-cpp)
+
+IBuffer::ExpectVoid FileBuffer::openTmpFile()
+{
+  const std::size_t countMax = 100;
+  std::size_t count = 0;
+  while (count < countMax) {
+    _fileName = _getRandomeFileName();
+    _fs.open(_fileName.c_str(), std::ios::in);
+    // check if file does not exit
+    if (!_fs.is_open()) {
+      // create file
+      _fs.open(_fileName.c_str(),
+               std::ios::in | std::ios::out | std::ios::binary |
+                 std::ios::trunc);
+      if (_fs.is_open()) {
+        _size = 0;
+        return ExpectVoid();
+      }
+    } else {
+      _fs.close();
+    }
+    ++count;
+  }
+  // tried to open 100 randome files
+  return ft::unexpected<BufferException>(errOpen);
+}
+
+/* ************************************************************************** */
+// PRIVATE
+
+IBuffer::ExpectStr FileBuffer::_getFront(std::size_t bytes)
+{
+  if (!_fs.is_open()) {
+    return ft::unexpected<BufferException>(errFileNotOpened);
+  }
+
+  std::string front(bytes, '\0');
+  _fs.read(&front[0], static_cast<std::streamsize>(bytes));
+  const std::streamsize actuallyRead = _fs.gcount();
+  if (actuallyRead == 0) {
+    return ft::unexpected<BufferException>(errFileEmpty);
+  }
+  front.resize(static_cast<std::size_t>(actuallyRead));
+  return front;
+}
+
+// NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+IBuffer::ExpectVoid FileBuffer::_saveRemainder()
+{
+  // Create a temporary file for the remainder
+  FileBuffer tmp;
+  const ExpectVoid resA = tmp.openTmpFile();
+  if (!resA.has_value()) {
+    return resA;
+  }
+
+  // Copy the remaining data
+  char buffer[_copyBufferSize];
+  while (_fs.good()) {
+    _fs.read(buffer, _copyBufferSize);
```

I think if the read failed it should also be an error and not silently ignored.